### ISSUE TITLE
[pending] NVDA reads the previous line in the editor unexpectedly when it is proceeded by a single newline

### ISSUE THREAD

**Issue by Neurrone** [Microsoft/vscode#41386]
<ul>
	<li>VSCode Version: Code - Insiders 1.20.0-insider (2bab83a097135cf8c2a1ca1498bb8906f06d2cb3, 2018-01-10T05:13:51.946Z)</li>
	<li>OS Version: Windows_NT x64 10.0.15063</li>
	<li><details><summary>Extensions (4)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-markdownlint|Dav|0.12.1
vscode-rust|kal|0.4.2
rust|rus|0.3.2
markdown-preview-enhanced|shd|0.3.2

</details></li>
<li> NVDA Version: master-14774,0d94f5dd</li>
</ul>

Steps to Reproduce:

1. Open a new file ctrl+n, and type or paste the following (including the trailing newline):

```
# test heading 1

```

*Actual behaviour*

If you type this, NVDA announces the line just typed ("# test heading 1") when pressing return. 

Now, press the down arrow key. It should correctly say blank since this is the last line, but reads the first line out instead. Pressing the left/right arrow keys to move by character reads as expected. Pressing up then down arrow, it still reads the second line incorrectly.

If you press enter again, the whole file reads properly as expected.

*Expected behaviour*

If typing this text manually, NVDA should not say anything on pressing return (as per default keyboard echo settings). The trailing newline should be read as "blank", instead of as the preceding line when using line navigation.

**Comment by alexandrudima** [comment]
@Neurrone I am sorry, but I could not reproduce.

I have created a file with precisely two lines, one with content, and one blank. Whenever I use arrow up (to go to the first line), NVDA reads out the first line. Whenever I use arrow down (to go to the second line), NVDA reads out "blank".

I have tried with NVDA 2018.1.1 and with VSCode Insiders.

**Comment by Neurrone** [comment]
@alexandrudima I was able to reproduce it, but no longer can after updating to the latest build. Maybe an upgrade to a later version of chrome fixed it. 

Thanks, closing.

**Comment by Neurrone** [comment]
Can reproduce again with latest insiders, noticed that the chrome version used was reverted from 60 to 58.

**Comment by alexandrudima** [comment]
Thank you @Neurrone . We had to roll back our Electron 2.0 upgrade, but we will attempt it again.

**Comment by bpasero** [comment]
I tried to reproduce and I do see a difference between our current insiders release and a release that uses Electron 2.0.x, but I am not sure the behavior is as described by @Neurrone 

**Code Insiders**
When I type enter to get to the newline, the first line is being read and not "blank".

**Code with Electron 2.0.x**
When I type enter to get to the newline, "blank" is being read (as expected).

@Neurrone can you try to reproduce again with latest NVDA and this build: [download](https://az764295.vo.msecnd.net/insider/50f90e42756f099fb14042e6bcf5d8d61e2b0a09/VSCode-win32-x64-1.26.0-insider.zip)

**Comment by bpasero** [comment]
**Update:** I tried from a real Windows desktop and now I actually do not see any difference between insiders with Electron `1.7.x` and `2.0.x`, in both cases here is what I do and what I get:
* type a line in a new untitled file
* enter to get a blank line
* arrow up and arrow down repeatedly and watch the "speech viewer" output from NVDA

=> in both cases when I am in the first line, that line is being read and when I am in the second blank line I see "blank" printed.

**Comment by Neurrone** [comment]
Hm, can still reproduce what I described with VS Code 1.26.0-insider (Commit: 6babef94104b27b252929324aa3c2d69cd38e9f2) and NVDA 2018.3 beta 2.

1. Type a line, press enter. That line is read, which isn't expected behaviour.
2. Arrow to the first line, then down to the second, blank line. It still announces the first line.

**Comment by bpasero** [comment]
@Neurrone thanks, that build is not using Electron 2.0.x, so this cannot be related to the update.

**Comment by isidorn** [comment]
@Neurrone can you please try reproducing this with latest vscode insiders and nvda?
I just tried with latest and could not reproduce. I wonder if I am doing something diferently than you

Here is my text. As I navigate using up / down though the lines I only get `blank` and the content of my line when I am exactly on my second line. So everything as expected

```

this is my second line

```

**Comment by Neurrone** [comment]
@isidorn this is fixed as of the latest insiders build - thanks!
